UI/GardenDeskForm.cs

```diff
@@ -101,27 +92,293 @@ private async void BtnLogin_Click(object sender, EventArgs e)
+            string lastName = IsInputBoxEmpty(textBoxLastName.Text, "The last name cannot be empty");
+            string email = IsInputBoxEmpty(textBoxEmailAddress.Text, "The e-mail address cannot be empty");
+            string phoneNumber = IsInputBoxEmpty(textBoxPhoneNumber.Text, "The phone number cannot be empty");
+            // first making sure the combo box is not empty then parsing it
+            // I could put a default selected text for combo box so you wouldn't have to check if it's empty
+            // but then the user could not notice that something is already selected and they could make a mistake
+            // and I'm designing for forgiveness here. I care about the user :(
+            string employeeRoleInput = IsInputBoxEmpty(comboBoxTypeUser.Text, "The employee role cannot be empty");
+            EmployeeRole employeeRole = (EmployeeRole)Enum.Parse(typeof(EmployeeRole), employeeRoleInput);
+            string branch = IsInputBoxEmpty(textBoxBranch.Text, "The branch cannot be empty");
+            return new Employee(firstName, lastName, email, phoneNumber, employeeRole, branch);
+        }
+
+        //  a method to make sure the textbox is not empty, if yes then throw an exception
+        private string IsInputBoxEmpty(string inputBoxString, string errorMessage)
+        {
+            if (inputBoxString == string.Empty || inputBoxString == null)
+                throw new Exception(errorMessage);
+
+            return inputBoxString;
+        }
+
+        // making sure that all text boxes are empty for a new employee
+        private void EmptyAllTextBox()
+        {
+            textBoxFirstName.Text = "";
+            textBoxLastName.Text = "";
+            comboBoxTypeUser.SelectedItem = null;
+            textBoxEmailAddress.Text = "";
+            textBoxPhoneNumber.Text = "";
+            textBoxBranch.Text = "";
+        }
+        #endregion
+        #region Edit employee
+
+        private void btnEditUser_Click(object sender, EventArgs e)
+        {
+            // set the selected employee to the current selected item in the list view
+            selectedEmployee = (Employee)usersList.SelectedItems[0].Tag;
+
+            // refresh the students panel so new information shows up
+            lblAddEditUser.Text = "Edit employee information";
+
+            // make the button for creating an employee invisible
+            btnCreateEmployee.Visible = false;
+
+            // the buttons and textboxes are changed before the panel is loaded to prevent them flickering after the panel loads
+            // make the buttons for editting an employee visible
+            btnDeleteEmployee.Visible = true;
+            btnUpdateEmployee.Visible = true;
+            // Prefill the text boxes with selected users' info
+            PrefillTextbox();
+
+            ShowPanel(pnlAddEditUser);
+        }
+        private async void btnUpdateEmployee_Click(object sender, EventArgs e)
```

Comment:
Most of your methods are crowded, please add new lines between methods for better readability.